src/pages/LoginPage.js

```diff
@@ -1,7 +1,98 @@
-import React from 'react'
+import React, { useRef, useState } from 'react'
+import { Heading, Flex, Box, Link, Button } from '@chakra-ui/react'
+import FormInput from '../components/UI/FormInput'
+import styles from '../styles/LoginPage.module.css'
+import { Link as ReactLink, useNavigate } from 'react-router-dom'
+import { loginUser } from '../api/auth'
+import { darken } from '@chakra-ui/theme-tools'
+import { getProfileData } from '../api/profile'
+import { useMultipleContexts } from '../hooks/multiple-context'
 
 const LoginPage = () => {
-   return <div>Login Page</div>
+   const emailInputRef = useRef()
+   const passwordInputRef = useRef()
+   const [errorLogin, setErrorLogin] = useState('')
+   const [loading, setLoading] = useState(false)
+   const navigate = useNavigate()
+   const { login, setUserProfile } = useMultipleContexts()
+
+   const handleFormSubmit = async (e) => {
+      e.preventDefault()
+      setLoading(true)
+
+      try {
+         const response = await loginUser({
+            identifier: emailInputRef.current.value.trim(),
+            password: passwordInputRef.current.value.trim(),
+         })
+
+         if (!response.ok) {
+            setErrorLogin(response.error)
+            setLoading(false)
+         } else {
+            const res = await getProfileData(response.user.id)
```

Comment:
We could move this call to auth context in a useEffect that gets triggered every time the jwt changes, that way we handle both the login case and the case when a logged in user opens the app and we have to load the profile for the first time